src/mpi/coll/algorithms/tree/types.h

```diff
@@ -0,0 +1,75 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/*
+ *  (C) 2006 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ *
+ *  Portions of this code were written by Intel Corporation.
+ *  Copyright (C) 2011-2017 Intel Corporation.  Intel provides this material
+ *  to Argonne National Laboratory subject to Software Grant and Corporate
+ *  Contributor License Agreement dated February 8, 2012.
+ */
+
+#include "coll_tree_types.h" /* from the ../common directory */
+#include "coll_args_types.h"
+
+typedef struct MPIR_ALGO_global_t {
+} MPIR_ALGO_global_t;
+
+typedef struct MPIR_ALGO_tree_comm_t {
+    int id; /* unique local id for the communicator */
+    MPIR_ALGO_tree_t tree;
+    int *curTag;
```

Comment:
Looks like the comm is missing from this struct, but these fields should just be in the `MPIR_Comm` anyway inside a union.